packages/shared/src/components/modals/streaks/StreakRecoverModal.spec.tsx

```diff
@@ -287,13 +322,17 @@ it('Should show success message on recover', async () => {
 
   // expect mutation to be called
   expect(mutationCalled).toBeTruthy();
+  await waitForNock();
```

Comment:
I think wait for nock might better be above since nock guarantees the above callback that sets `mutationCalled` has been executed?

Reply:
or you can use `waitFor(() => expect(mutationCalled).toBeTruthy())`

Reply:
I've tried right now changing this but time to time test are failing again :(

Reply:
I think different one failed right now on ci? https://app.circleci.com/pipelines/github/dailydotdev/apps/19713/workflows/3c038725-d05f-43b1-82d7-adc25c032203/jobs/51077

Reply:
Yes, it is not from the recovery but I can  also quickly check for it 👀

Reply:
Let's keep it scoped, I was just referring to you saying it failed. So the test you are fixing is failing locally then?

Reply:
Yes, if I remove `await waitForNock();` and put `waitFor(() => expect(mutationCalled).toBeTruthy())` is failing locally once in a while

Reply:
not remove it but reverse order:
```
await waitForNock();
expect(mutationCalled).toBeTruthy();
```

Reply:
Nope, not working :\ 
But anyway before this line there is another `waitForNock` so we have one before and one after!

Reply:
And its working when nock is after?

Reply:
I see its two waitForNock now, do you know why we need second one?

Reply:
I think that is because of multiple mutation have been called before. I've tried to change it like this:

```ts
let mutationCalled = false;
  mockRecoveryMutation({}, () => {
    mutationCalled = true;
  });
  await waitForNock();

  mockAlertsMutation();
  await waitForNock();

  // expect mutation to be called
  await waitFor(() => expect(mutationCalled).toBeTruthy());

```

and seems working, I think is easier to understand and maintain, wdyt?

Reply:
Nevermind, if I remove the wait for nock after expect(mutationCalled) is failing time to time, had to restore it. :\

Reply:
From my perspective you would need only one waitForNock at the end, but flakiness is hard. 

`mockRecoveryMutation`  and `mockAlertsMutation` are just handlers that get setup and wait for requests so they should go first before any actions in the tests. 

I don't see the reason why we would call them incrementally one by one and then wait for them, thats why I asked if you know why its multiple waitForNock calls?

Reply:
I've tried to remove some, but component do some api call one after another and I think that some more waiting is needed for this reason?

Reply:
so the flow of the test how I see it:

1. mock recovery query
2. renderComponent -> this consumes above mock
3. waitForNock (to make sure we wait for above consumes to go through)
4. mock mutation and alerts mutation 
5. fireEvent.click (click on recover) -> this consumes above mock
6. waitForNock (to make sure we wait for above consumes to go through) (this fails 🔴 )
7. evaluate mutation called
8. evaluate toast

In your solution you have 4. and 5. reversed. But it still fails with both so I think the root cause is something else. From what I can se You can run test multiple times by replace `it()` with following:
```js
test.each(Array(100).fill(undefined))(
  'Should show success message on recover',
  async () => {
    // test code
  });
```

it takes a bit of time but it will show if test is flaky.

Reply:
This was my approach, but above works better most cases as concurrently will break almost all our test anyway.

https://dailydotdev.slack.com/archives/C02E2C3C13R/p1723804495112589